ethereum/ethevents/events.go

```diff
@@ -290,13 +290,36 @@ func (ev *EthereumEvents) SubscribeEthereumEventLogs(ctx context.Context, fromBl
 		go ev.runEventProcessor(ctx)
 	}
 
+	// check eth connection
+	connCh := make(chan bool, 1)
+	go func(chan bool) {
+		for {
+			time.Sleep(time.Second * 5)
+			blockTctx, cancel := context.WithTimeout(ctx, types.EthereumReadTimeout*2)
+			if _, err := ev.VotingHandle.EthereumClient.BlockByNumber(blockTctx, nil); err != nil {
+				log.Warnf("cannot check ethereum connection while running the event processor, %s", err)
+				connCh <- false
+				cancel()
```

Comment:
cancel() should also be called if not error

Reply:
moved and added a defer that will be executed in the very end